### ISSUE TITLE
Use groovy docker image for running tests in CI

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#12646]
Currently, we are using apt repository (http://azure.archive.ubuntu.com/ubuntu) to fetch dependencies for installing groovy.

With the increase in the recent amount of failures (https://github.com/checkstyle/checkstyle/actions/runs/3909700454/jobs/6681066344) migrating to reliable source is important.

Using [docker](https://hub.docker.com/_/groovy) to run the scripts would result in the following benefits:
1. Flexibility to upgrade or downgrade versions (v3 and above).
2. Reliability as docker hub is rarely down.

**Comment by Vyom-Yadav** [comment]
@nrmancuso @rnveach @romani Please share your opinion.

**Comment by nrmancuso** [comment]
Why can’t we use our Java/groovy/mvn image that we already have?

**Comment by Vyom-Yadav** [comment]
> Why can’t we use our Java/groovy/mvn image that we already have?

We have one, please share the link

> One example of a failed run does not constitute an increase in the amount of failures :)

I just linked a single run but this failed at least 10 times

**Comment by nrmancuso** [comment]
We use our custom image in a bunch of CircleCI executions, here is the link: https://hub.docker.com/r/checkstyle/jdk-11-groovy-git-mvn

**Comment by Vyom-Yadav** [comment]
It uses groovy 3.0.9, shouldn't we update it to use the latest stable version, same for others.

**Comment by nrmancuso** [comment]
Makes sense to me to update to latest patch version, please create an issue in contribution repo.

**Comment by rnveach** [comment]
There is something similar at https://github.com/checkstyle/contribution/issues/671

**Comment by Vyom-Yadav** [comment]
Opened https://github.com/checkstyle/contribution/issues/763
